server/internal/controllers/chain.go

```diff
@@ -379,34 +380,13 @@ LIMIT 1
 			c.String(http.StatusInternalServerError, "User could not be added to chain due to unknown error")
 			return
 		}
-
-		// find admin users related to the chain to email
-		results, err := models.UserGetAdminsByChain(db, chain.ID)
+		err := services.EmailLoopAdminsOnUserJoin(db, user, chain.ID)
 		if err != nil {
-			goscope.Log.Errorf("Error retrieving chain admins: %s", err)
-			c.String(http.StatusInternalServerError, "No admins exist for this loop")
-			return
-		}
-		if len(results) == 0 {
-			goscope.Log.Errorf("Empty chain that is still public: ChainID: %d", chain.ID)
-			c.String(http.StatusInternalServerError, "No admins exist for this loop")
+			goscope.Log.Errorf("Unable to send email to associated loop admins: %v", err)
+			c.String(http.StatusInternalServerError, err.Error())
 			return
 		}
-
-		for _, result := range results {
-			if result.Email.Valid {
-				go views.EmailSomeoneIsInterestedInJoiningYourLoop(db, result.I18n,
-					result.Email.String,
-					result.Name,
-					chain.Name,
-					user.Name,
-					user.Email.String,
-					user.PhoneNumber,
-					user.Address,
-					user.Sizes,
-				)
-			}
-		}
+		services.EmailYouSignedUpForLoop(db, user, chain.Name)
```

Comment:
also no error handled here.

Reply:
also: this function does not seem to be complete: the templates for "you_signed_up_for_loop" are missing?